innopoints/models/activity.py

```diff
@@ -36,7 +36,13 @@ class Activity(db.Model):
     """Represents a volunteering activity in the project."""
     __tablename__ = 'activities'
     __table_args__ = (
-        db.CheckConstraint(f'(fixed_reward AND working_hours = 1) '
+        db.CheckConstraint('working_hours == NULL OR working_hours >= 0',
+                           name='working hours are non-negative'),
+        db.CheckConstraint('people_required == NULL OR people_required > 0',
+                           name='people required are unset or positive'),
+        db.CheckConstraint('NOT draft AND working_hours != NULL',
+                           name='working hours are not nullable for non-drafts'),
+        db.CheckConstraint(f'NOT draft AND (fixed_reward AND working_hours = 1) '
```

Comment:
Same here. will fail if it is a draft and `fixed_reward`.
Note that `NOT` has a higher precedence than `AND` (https://www.postgresql.org/docs/12/sql-syntax-lexical.html#SQL-PRECEDENCE-TABLE)